src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,45 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type == Bulldozer.class) {
+            BulldozerProducer bulldozerProducer = new BulldozerProducer();
+            return new ArrayList<>(bulldozerProducer.get());
+        }
+        if (type == Excavator.class) {
+            ExcavatorProducer excavatorProducer = new ExcavatorProducer();
+            return new ArrayList<>(excavatorProducer.get());
+        }
+        if (type == Truck.class) {
+            TruckProducer truckProducer = new TruckProducer();
+            return new ArrayList<>(truckProducer.get());
+        }
+        return null;
```

Comment:
```suggestion
        return Collections.EMPTY_LIST;
```
it's better to return empty list instead to avoid NPE